src/Ormolu/Exception.hs

```diff
@@ -35,17 +36,28 @@ instance Exception OrmoluException where
       ]
     OrmoluParsingFailed s e ->
       showParsingErr "Parsing of source code failed:" s e
-    OrmoluOutputParsingFailed s e ->
+    OrmoluOutputParsingFailed w s e ->
       showParsingErr "Parsing of formatted code failed:" s e ++
+        (maybe "" (\sw -> showSemicolonWarning sw ++ "\n") w) ++
         "Please, consider reporting the bug.\n"
-    OrmoluASTDiffers path ss -> unlines $
+    OrmoluASTDiffers w path ss -> unlines $
       [ "AST of input and AST of formatted code differ."
       ]
       ++ (fmap withIndent $ case fmap (\s -> "at " ++ showOutputable s) ss of
            [] -> ["in " ++ path]
-           xs -> xs) ++
-      [ "Please, consider reporting the bug."
-      ]
+           xs -> xs)
+      ++ maybe [] (pure . showSemicolonWarning) w
+      ++ [ "Please, consider reporting the bug."
+         ]
+
+data SemicolonWarning = SemicolonWarning [Int]
```

Comment:
Documentation would be good here. In general, even type and most functions can benefit from documentation.